### PR TITLE
Bit.Country's General Grant Application

### PR DESCRIPTION
**Pull request by bit-country**
## Grant Application

This application is (select one):
- [x] Speculative (use this by default)
- [ ] an RFP response

###

This application is (select one):
- [ ] Public (fully)
- [x] Public with private finances

### Abstract

The project team is currently in the process of applying for the Substrate Builders Program.

- Bit.Country is a decentralized world. Putting the community first. 
- Bit.Country chain is an NFT/game protocol for managing & incentivizing small-medium communities using gamification.
- In Bit.Country, you can create your own space, communities, economics, and digital assets (NFTs). 
- The dapp offers both a classical web view and a 3D in-browser view of the country. 
- The decentralized marketplace allows users to trade their digital assets (NFTs) with each other.
- Piloted the concept with 109 active users and observed a great level of user engagement.

### POC Preview
* View the POC on Medium [Screens and Video](https://medium.com/@bitcountry/create-and-live-in-a-decentralized-world-of-your-own-4ad872379d00)
* View the POC Dapp [https://bit.country](https://bit.country)

## Checklist
- [x] The [grants](https://github.com/w3f/Web3-collaboration/blob/master/grants/grants.md) document has been read and understood.
- [x] The [Google Form](https://docs.google.com/forms/d/e/1FAIpQLSfMfjiRmDQDRk-4OhNASM6BAKii7rz_B1jWtbCPkUh6N7M2ww/viewform) will be completed accurately. Note that the Google Form requires the pull request URL.
- [x] Abstract (above) is succinct and complete.
- [x] The application is being included into the correct directory: either '[targeted](https://github.com/w3f/Web3-collaboration/tree/master/grants/targeted)' or '[speculative](https://github.com/w3f/Web3-collaboration/tree/master/grants/speculative)'.
- [x] The application includes a project description.
- [x] The application includes all names of team members.
- [x] The application includes a description of the team's experience.
- [x] The application includes all necessary links (e.g. GitHub and LinkedIn)
- [x] The "Development Roadmap" section in the application has a timeline of development ("milestones").
- [ ] (Private)The "Development Roadmap" section in the application has an estimate of funds required.
- [x] The "Development Roadmap" section gives an indication of the team's long term plans.
- [x] The "Development Roadmap" section includes [documentation](https://github.com/w3f/Web3-collaboration/blob/master/grants/grants.md#documentation) as a deliverable for at least one milestone.

### CODE REVIEW

**Comment by mmagician** [comment]
@bit-country Thanks for your application. While you provide a high level overview, the milestones are lacking technical specifications:
- How do you plan to serve/distribute your Dapp? Is http://bit.country going to be the place where it's hosted (IPFS?), or maybe you want to have mobile apps / others?
- The substrate modules would need some more details, like what public of functions would be callable & what functionality they'd preform, what storage it'll have etc.
- Could you clarify why you need 5 FTE for each milestone for 4 weeks? Perhaps your deliverables are not detailed enough, but my feeling is that similar scope (creating 1 pallet per milestone + some frontend) have been achieved in the past by 1-2 person teams within shorter timeframes than 4 weeks/milestone.

Also, regarding the ecosystem fit, perhaps you might want to have a look at [Upland](https://upland.me/). I've "played" it for some time & got bored very quickly - there was little to do in the game besides spending fiat money. While a concept of owning digital assets (in their case property, in yours country/residence/NFTs) might appeal to some users, my personal opinion is that calling & marketing it as a "game" is a misnomer. 
Then, Upland's USP was owning virtual property corresponding to real-world places. Not sure why I'd want to own a share of a made-up country.

**Comment by bit-country** [comment]
Hi @mmagician ,

Thanks very much for your rapid response and feedback.

>While a concept of owning digital assets (in their case property, in yours country/residence/NFTs) might appeal to some users, my personal opinion is that calling & marketing it as a "game" is a misnomer.

You are right, we just recently added the "game" keyword to the application yesterday, in fact, I see bit.country is a new way of social and managing a community. Not only connecting people and creating posts on the timeline but also people can meet each other in the 3D Voxel world within the same country and block. This creates another dimension of interactions among users with common interests.

Currently, we're piloting it with 109 active users from our network, they are using it as a social tool e.g. share knowledge for a topic and blogging. We presented our idea of making the 3D view available for blocks so everyone can see each other walk around and interact behind the timeline view, the response from our user group is quite positive.

Bit.country provides the functions and decentralization governance for users to co-manage and thrive a community, we have seen a lot of use cases. We use it to manage tech students to interact with each other especially in this uncertain time. We believe any business that wants to look after or grow a community can use bit.country to achieve their goals.


> @bit-country Thanks for your application. While you provide a high level overview, the milestones are lacking technical specifications:
> 
> * How do you plan to serve/distribute your Dapp? Is http://bit.country going to be the place where it's hosted (IPFS?), or maybe you want to have mobile apps / others?

Our dapp bit.country will be hosted on the cloud and our chain specific logic will be built into the Substrate pallet modules. In order to introduce more users to learn and onboard to blockchain platform, we would like to create no barrier for anyone to register bit.country. By default each bit.country created is off-chain, the country admin needs to upgrade it to be on-chain. I will describe more in the application.

> * The substrate modules would need some more details, like what public of functions would be callable & what functionality they'd preform, what storage it'll have etc.
Thanks for your advice @mmagician . I will add more details today.

> * Could you clarify why you need 5 FTE for each milestone for 4 weeks? Perhaps your deliverables are not detailed enough, but my feeling is that similar scope (creating 1 pallet per milestone + some frontend) have been achieved in the past by 1-2 person teams within shorter timeframes than 4 weeks/milestone.
From the details I provided,  it is aligned with what you mentioned. I'll provide more details.

Again, thanks for your quick response and valuable feedback. I will add more details and let you know once it is done.

Best Regards,
Ray Lu


> @bit-country Thanks for your application. While you provide a high level overview, the milestones are lacking technical specifications:
> 
> * How do you plan to serve/distribute your Dapp? Is http://bit.country going to be the place where it's hosted (IPFS?), or maybe you want to have mobile apps / others?
> * The substrate modules would need some more details, like what public of functions would be callable & what functionality they'd preform, what storage it'll have etc.
> * Could you clarify why you need 5 FTE for each milestone for 4 weeks? Perhaps your deliverables are not detailed enough, but my feeling is that similar scope (creating 1 pallet per milestone + some frontend) have been achieved in the past by 1-2 person teams within shorter timeframes than 4 weeks/milestone.
> 
> Also, regarding the ecosystem fit, perhaps you might want to have a look at [Upland](https://upland.me/). I've "played" it for some time & got bored very quickly - there was little to do in the game besides spending fiat money. While a concept of owning digital assets (in their case property, in yours country/residence/NFTs) might appeal to some users, my personal opinion is that calling & marketing it as a "game" is a misnomer.
> Then, Upland's USP was owning virtual property corresponding to real-world places. Not sure why I'd want to own a share of a made-up country.

**Comment by bit-country** [comment]
We've created a Medium article with video and screens of the POC.
[Click here to view more screenshots and video](https://medium.com/@bitcountry/create-and-live-in-a-decentralized-world-of-your-own-4ad872379d00)

**Comment by mmagician** [comment]
@bit-country Thanks for the updates. You still haven't clarified why you need a 5-person team for each of the milestones. This seems like a lot of people for the amount of work you propose.

**Comment by bit-country** [comment]
@mmagician Thanks for your valuable time and review. Please let me explain why we need a 5-person team for each of the milestones. To make it easier to read, I have divided the content into sections as below,

**The Goal**
After the 3-months plan, our final goal is to have a complete MVP solution (Both the app and chain) and onboard the rest of our network members (~1400 users remaining). Then, we will run our marketing campaign to onboard more users outside of the platform.  

At this stage, we budgeted (from our own) and targeted to onboard 50K users starting from Feb 2021 (Last milestone) to Aug 2021.

**Meet the Scale**
To meet the required scale, we will need a production-ready & full-fledged web app & infrastructure.  Especially our web app, we design it the way it can work off-chain first. Hence, people have no barrier to creating their communities; this way, we can help onboard users more quickly. We will then introduce the decentralization concept by encouraging them to upgrade their bit countries or assets to be on-chain.

**Support the Scale** 
To build bit.country web app (dapp) that potentially serves 100 concurrent users will need to create more sophisticated architecture and features. 

We will need to work on the following behind the scene during the three months apart from the Substrate modules.

- Utilising the caching layer for serving content, key security attributes, country/block/section membership and carefully refreshing the cache by avoid caching stampede and invalidity of data. We do have a separate database and caching server behind our web dapp.
- Writing background processors to prepare aggregation of data for insights, personalized timeline content, incentive rules and some of the marketplace rules.
- Creating the security module to ensure content is shared with the intended audience. e.g. a post can be within one or many block context, country context, public or completely private. 
- Creating the features allow country owners to perform the "Upgrade Your Bit Country to Blockchain" process. It is important to ensure the classic internet users will be educated in this process via our web app, and have a pleasing and comfortable experience with decentralized technologies to appreciate the impact that the blockchain can bring without any barriers.
- To guide the users, we need to create graphics, videos, and documentation to illustrate how it works.
- We will need to create Cypress automated testing to cover most of our dapp features.
- We need to utilize our marketing team and community management team to run campaigns and onboard the users.

**More Discovery with Substrate Builders Program**
Our team had been invited to meet the team at Substrate Builders Program to explore the project further. 

We will have our next meeting on the 28th this month; I can update you on the outcome and understanding of our runtime modules. There are some more pallets we want to include in the protocol, but we haven't documented them in the application as to it is best to add it to the specification after we talked to the specialized team. 

**Teams**
As you can see, apart from our 5 FTE, I have also included three other part-timers from different departments of our group. In reality, we may need more people to get involved to take off this project eventually. 

We're quite serious about this project and believe it has the potential to become a unicorn in the area of managing & incentivizing small-medium community.

After my research, I am aware that the grant may only focus on building & innovating the substrate runtime modules; 

However, to better reveal the Substrate and Polkadot network's power and features to classic internet users, we believe we need to build a decent web app or dapp to attract them and keep them engaged on the UI/UX side. And that is what we are trying to do here within the scope of the milestones.

The direct benefit we invest our resources on the web app (dapp) is that if our goal is reached, we could bring many classic internet users to learn and understand the decentralized paradigm, Substrate, and Polkadot ecosystem.

If the grant team thinks the application should only focus on the runtime modules, I will make adjustments and allocate 1 or 2 people for that part on the project.  We're happy to follow the team's advice and suggestions.

In summary, to achieve what we planned, I do feel we need a 5-FTE team at this stage. 

**TL;DR**

- We plan to build a full-fledged solution that is ready to bring 1400 internal users and 50K external users by Aug 2021
- Our dapp requires truck load of development and testing to ensure we attract classic internet users.
- We will explore more with Substrate Builders Programme team to include more modules in our design.
- We are serious about the project, and believe it will bring many people to the ecosystem.

Thanks for your time reading everything we plan to do. I hope it is clear and makes more sense now.

Best Regards,
Ray Lu
Founder.

**Comment by mmagician** [comment]
@bit-country Thanks for the updates. I would be curious to hear about your learnings from the Builders Program.

> After my research, I am aware that the grant may only focus on building & innovating the substrate runtime modules;

In general we may fund small components of a project other than just the substrate modules, if they help to onboard the users. I see that in your case, however, substrate is only a small element compared to other changes your application needs in order to scale. 

If you want to try to get the grant for 5 FTEs, you'd need to include all the relevant technical details of what you plan to do "behind the scenes" in your application. **At the moment none of that is not reflected in your milestones.** A rule of thumb: the more specifications you provide, the higher your chances of a positive outcome.

**Comment by bit-country** [comment]
@mmagician Thanks for your insightful feedback. And yes we strategized our application for onboardings users to the ecosystem & Blockchain. That needs a full-fledged web app (dapp) and strong backing of decentralized technology (Luckily Substrate is here).

Previously, we only focused on putting runtime modules related to the application as I thought to keep its Substrate focus and simple, in this case, we will add all the Dapp specifications that we planned to the application. I will let you know once it is done. Have a good weekend.

Regards, 
Ray

**Comment by bit-country** [comment]
@mmagician Trust you are well. As per your advice, we've added a lot more details to the specs. Also added some images and video in the application. Please let us know if there is anything else we need to do. Here is the [shortcut to the application.](https://github.com/bit-country/General-Grants-Program/blob/master/grants/speculative/bit_country.md)

Thanks & Regards,
Ray
